lambda.tf

```
@@ -0,0 +1,157 @@
+######################################################################
+## https://docs.datadoghq.com/integrations/amazon_rds/?tab=enhanced
+
+locals {
+  lambda_enabled         = module.this.enabled && var.dd_api_key_source.resource != "" ? true : false
+  dd_api_key_resource    = var.dd_api_key_source.resource
+  dd_api_key_identifier  = var.dd_api_key_source.identifier
+  dd_api_key_arn         = local.dd_api_key_resource == "ssm" ? data.aws_ssm_parameter.api_key[0].arn : local.dd_api_key_identifier
```

```suggestion
  dd_api_key_arn         = local.dd_api_key_resource == "ssm" ? join("", data.aws_ssm_parameter.api_key.*.arn : local.dd_api_key_identifier
```

If `lambda_enabled=false`, it will complain about the list not having any items